spec/motion/revision_calculator_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+RSpec.describe Motion::RevisionCalculator do
+  subject(:calculator) do
+    described_class.new(
+      revision_paths: revision_paths
+    )
+  end
+
+  let(:revision_paths) { Rails.application.config.paths.dup }
+
+  describe "#perform" do
+    subject(:output) { calculator.perform }
+    let(:empty_hash) { Digest::MD5.new.hexdigest }
+
+    context "when the revisions path is not a Rails::Paths::Root object" do
+      let(:revision_paths) { [] }
+
+      it "raises BadRevisionPathsError" do
+        expect { subject }.to raise_error(Motion::BadRevisionPathsError)
+      end
+    end
+
+    context "when there are no paths to hash" do
+      let(:revision_paths) { Rails::Paths::Root.new(Rails.application.root) }
+      let(:empty_hash) { Digest::MD5.new.hexdigest }
+
+      it "hashes empty digest" do
+        expect(subject).to eq(empty_hash)
+      end
+    end
+
+    context "when paths do not exist" do
+      let(:revision_paths) do
+        paths = Rails::Paths::Root.new(Rails.application.root)
+        paths.add "foo"
+        paths
+      end
+
+      it "ignores empty directory" do
+        expect(subject).to eq(empty_hash)
+      end
+    end
+
+    context "for normal application with files" do
+      let(:all_dirs) { revision_paths.all_paths.flat_map(&:existent) }
+      let(:first_dir) { all_dirs.first }
+      let(:new_file) { "#{first_dir}/foot.txt" }
+
+      before(:each) { File.delete(new_file) if File.exist?(new_file) }
+      after(:each) { File.delete(new_file) if File.exist?(new_file) }
```

Comment:
Can we use an `around(:each)` with a temporary directory for this? It will make sure everything gets cleaned up for us:

```ruby
attr_reader :tempdir

around(:each) do |example|
  Dir.mktmpdir do |path|
    @tempdir = path
 
    example.run
  end
end
```